### PR TITLE
[SPARK-13324] [CORE] [BUILD] Update plugin, test, example dependencies for 2.x

### PR DESCRIPTION
**Pull request by srowen**
Phase 1: update plugin versions, test dependencies, some example and third-party versions

### CODE REVIEW

**Comment by srowen** [comment]
Jenkins, retest this please

File: build/mvn [hunk 1]
```diff
@@ -69,7 +69,7 @@ install_app() {
 
 # Install maven under the build/ folder
 install_mvn() {
-  local MVN_VERSION="3.3.3"
+  local MVN_VERSION="3.3.9"
```
Comment (JoshRosen) [inline build/mvn]: If we're going to bump Maven, should we also make sure that we have an enforcer rule which doesn't allow any earlier version of Maven to be used?

Comment (srowen) [inline build/mvn]: It's already in the enforcer rule, but oops, I did not update the expected Maven version in the parent POM. Coming now ...

**Comment by JoshRosen** [comment]
These all look like pretty low risk version bumps to me, so I'm going to merge this into master. Thanks Sean!

**Pull request closed without merging**